api/src/services/listCourtCases.ts

```diff
@@ -0,0 +1,214 @@
+import type { DataSource, SelectQueryBuilder } from "typeorm"
+import { Brackets, ILike, IsNull, LessThanOrEqual, MoreThan, MoreThanOrEqual, Not } from "typeorm"
+import type { CaseListQueryParams } from "../types/CaseListQueryParams"
+import type { ListCourtCaseResult } from "../types/ListCourtCasesResult"
+import type PromiseResult from "../types/PromiseResult"
+import { isError } from "../types/Result"
+import { BailCodes } from "../utils/bailCodes"
+import CourtCase from "./entities/CourtCase"
+import courtCasesByVisibleForcesQuery from "./queries/courtCasesByVisibleForcesQuery"
+import Note from "./entities/Note"
+
+const listCourtCases = async (
+  connection: DataSource,
+  {
+    pageNum,
+    maxPageItems,
+    forces,
+    orderBy,
+    order,
+    defendantName,
+    courtName,
+    ptiurn,
+    reasons,
+    urgent,
+    courtDateRange,
+    locked,
+    caseState,
+    allocatedToUserName,
+    reasonCode,
+    resolvedByUsername
+  }: CaseListQueryParams
+): PromiseResult<ListCourtCaseResult> => {
+  const pageNumValidated = (pageNum ? parseInt(pageNum, 10) : 1) - 1 // -1 because the db index starts at 0
+  const maxPageItemsValidated = maxPageItems ? parseInt(maxPageItems, 10) : 25
+  const repository = connection.getRepository(CourtCase)
+  const subquery = connection
+    .getRepository(Note)
+    .createQueryBuilder("notes")
+    .select("COUNT(note_id)")
+    .where("error_id = courtCase.errorId")
+  let query = repository.createQueryBuilder("courtCase")
+  query = courtCasesByVisibleForcesQuery(query, forces) as SelectQueryBuilder<CourtCase>
```

Comment:
FYI there were [some changes](https://github.com/ministryofjustice/bichard7-next-ui/pull/243/files) to `courtCasesByVisibleForcesQuery`, and also added filter for [excluded triggers.](https://github.com/ministryofjustice/bichard7-next-ui/pull/250/files). Maybe not in this, but It would be good to update this in a follow up PR!